gordon/interfaces.py

```diff
@@ -73,52 +88,26 @@ def __init__(config, success_channel, error_channel, metrics=None):
             metrics (obj): Optional obj used to emit metrics.
         """
 
-    async def shutdown():
-        """Gracefully shutdown plugin."""
-
-
-class IEventConsumerClient(IGenericPlugin):
-    """Client for ingesting push/pull events for Gordon to process.
-
-    The client also receives both successful and failed
-    :py:class:`gordon.interfaces.IEventMessage` objects from Gordon in order
-    to perform cleanup if needed.
-    """
-
     async def run():
         """Begin consuming messages using the provided event loop."""
 
-    async def cleanup(event_msg):
-        """Perform cleanup tasks related to a message.
-
-        Args:
-            event_msg (IEventMessage): Message at the end of its life cycle.
-        """
 
+class IMessageHandler(IGenericPlugin):
+    """Plugin which performs some operation on an event message."""
```

Comment:
I wonder if it's worth mentioning that the router within core will route messages from/to whomever implements a message handler. I think the sentence you have here is perfect, so maybe start another line (like how the old IEventConsumerClient had).